### DISCUSSION TITLE
What is the main proper of the ErrorSet class?

### DISCUSSION THREAD

**Discussion by nguyenvantruong17** [gialynguyen/vcc-schema#11] [Q&A] [answered]
![Screen Shot 2021-06-01 at 23 48 41](https://user-images.githubusercontent.com/18642063/120361380-f79de280-c333-11eb-9df0-d161022f1802.png)

**Comment by nguyenvantruong17** [comment]
hasPrerequisiteError property dùng cho mục đích gì nhỉ?

**Comment by nguyenvantruong17** [comment]
Tại sao mỗi lần class được khởi tạo lại phải lấy prototype để gán ngược lại cho nó khi khởi tạo.
Prototype instance ở đây có phải là của Error built-in javascript
![Screen Shot 2021-06-01 at 23 51 07](https://user-images.githubusercontent.com/18642063/120361620-40559b80-c334-11eb-88cb-998bc3af88c8.png)

**Reply by gialynguyen** [reply]
đúng v

**Comment by gialynguyen** [comment, marked as answer]
là 1 data structure cho các ErrorSubject trong quá trình parse diễn ra.

**Reply by nguyenvantruong17** [reply]
Vậy thì khi nên nên dùng ErrorSet?

**Reply by gialynguyen** [reply]
Thật ra nó không phải phải là 1 thứ sinh ra để sử dụng, nó thuộc về core thì đúng hơn. Nhưng nếu muốn sử dụng, thì nó được sử dụng khi muốn collect các error, lúc này sẽ có sẵn các method như: get message, format error to object like...

**Reply by nguyenvantruong17** [reply]
Vậy tại sao nó lại được sử dụng như 1 Return Type của Checker. Chỗ đó thật sự confuse. Nếu không đụng tới thì nên loại bỏ đi

**Reply by gialynguyen** [reply]
Nó là Return Type của quá trình parse sau khi diễn ra. 1 quá trình parse diễn ra sẽ sinh ra nhiều ErrorSubject, và ErrorSet nó sẽ quản lý các ErrorSubject của quá trình đó
